src/dymaptic.GeoBlazor.Core/Scripts/arcGisJsInterop.ts

```diff
@@ -305,36 +305,8 @@ export async function buildMapView(id: string, dotNetReference: any, long: numbe
         disposeView(id);
         let view: View;
 
-        let basemap: Basemap | undefined = undefined;
-        let basemapLayers: any[] = [];
-        if (!mapType.startsWith('web')) {
-            if (mapObject.arcGISDefaultBasemap !== undefined &&
-                mapObject.arcGISDefaultBasemap !== null) {
-                basemap = mapObject.arcGISDefaultBasemap;
-            } else if (hasValue(mapObject.basemap?.style)) {
-                let style = new BasemapStyle({
-                    id: mapObject.basemap.style.name
-                });
-                if (hasValue(mapObject.basemap.style.language)) {
-                    style.language = mapObject.basemap.style.language
-                }
-                if (hasValue(mapObject.basemap.style.serviceUrl)) {
-                    style.serviceUrl = mapObject.basemap.style.serviceUrl;
-                }
-                basemap = new Basemap({ style: style })
-            } else if (hasValue(mapObject.basemap?.portalItem?.id)) {
-                let portalItem = buildJsPortalItem(mapObject.basemap.portalItem);
-                basemap = new Basemap({ portalItem: portalItem });
-            } else if (mapObject.basemap?.layers.length > 0) {
-                for (let i = 0; i < mapObject.basemap.layers.length; i++) {
-                    const layerObject = mapObject.basemap.layers[i];
-                    basemapLayers.push(layerObject);
-                }
-                basemap = new Basemap({
-                    baseLayers: []
-                });
-            }
-        }
+        let basemap = getBaseMap(mapObject, mapType);
+        let basemapLayers = getBasemapLayers(mapObject, mapType);
```

Comment:
In the `develop` version, we only collect basemap layers if there isn't a basemap style or portal item. With this refactor, would it be possible for someone to declare both a style and extra layers, and does ArcGIS support this? I can do some investigating this weekend, but I think we should check before we merge this.

Reply:
Yes, I believe you can.